File: app/src/main/java/hpn332/cb/model/database/DBHelper.java
package hpn332.cb.model.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

class DBHelper extends SQLiteOpenHelper {

	private static final String TAG = "DBHelper";

	private static final String DATABASE_NAME    = "db_calender.db";
	private static final int    DATABASE_VERSION = 1;

	DBHelper(Context context) {
		super(context, DATABASE_NAME, null, DATABASE_VERSION);
	}

	@Override
	public void onCreate(SQLiteDatabase db) {

		db.execSQL(DBContract.ProjectEntry.CREATE_TABLE);
		db.execSQL(DBContract.TaskEntry.CREATE_TABLE);
		db.execSQL(DBContract.TagEntry.CREATE_TABLE);
		db.execSQL(DBContract.BackLogEntry.CREATE_TABLE);
	}

	@Override
	public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

		Log.w(TAG, "Upgrading database from version " + oldVersion + " to "
				+ newVersion + ", which will destroy all old data");

		if (oldVersion < newVersion) {
			db.execSQL("DROP TABLE IF EXISTS " + DBContract.TABLE_PROJECT);
			db.execSQL("DROP TABLE IF EXISTS " + DBContract.TABLE_TASK);
			db.execSQL("DROP TABLE IF EXISTS " + DBContract.TABLE_TAG);
			db.execSQL("DROP TABLE IF EXISTS " + DBContract.TABLE_BACKLOG);

			onCreate(db);
		}
	}

	static void deleteDatabase(Context context) {
		context.deleteDatabase(DATABASE_NAME);
	}
}

File: app/src/main/java/hpn332/cb/model/data/DBHelper.kt
package hpn332.cb.model.data

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.util.Log

internal class DBHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

	override fun onCreate(db: SQLiteDatabase) {

		db.execSQL(DBContract.ProjectEntry.CREATE_TABLE)
		db.execSQL(DBContract.TaskEntry.CREATE_TABLE)
		db.execSQL(DBContract.TagEntry.CREATE_TABLE)
		db.execSQL(DBContract.BackLogEntry.CREATE_TABLE)
	}

	override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {

		Log.w(TAG, "Upgrading database from version " + oldVersion + " to "
				+ newVersion + ", which will destroy all old data")

		if (oldVersion < newVersion) {
			db.execSQL("DROP TABLE IF EXISTS " + DBContract.TABLE_PROJECT)
			db.execSQL("DROP TABLE IF EXISTS " + DBContract.TABLE_TASK)
			db.execSQL("DROP TABLE IF EXISTS " + DBContract.TABLE_TAG)
			db.execSQL("DROP TABLE IF EXISTS " + DBContract.TABLE_BACKLOG)

			onCreate(db)
		}
	}

	companion object {

		private val TAG = "DBHelper"

		private val DATABASE_NAME = "db_calender.db"
		private val DATABASE_VERSION = 1

		fun deleteDatabase(context: Context) {
			context.deleteDatabase(DATABASE_NAME)
		}
	}
}
